src/SonarQube.Client/Models/SonarQubeIssue.cs

```diff
@@ -22,24 +22,24 @@ namespace SonarQube.Client.Models
 {
     public class SonarQubeIssue
     {
-        public SonarQubeIssue(string filePath, string hash, int? line, string message, string moduleKey,
-            SonarQubeIssueResolutionState resolutionState, string ruleId)
+        public SonarQubeIssue(string filePath, string hash, int? line, string message, string moduleKey, string ruleId, 
+            string status)
         {
             FilePath = filePath?.Trim('/', '\\');
             Hash = hash;
             Line = line;
             Message = message;
             ModuleKey = moduleKey;
-            ResolutionState = resolutionState;
             RuleId = ruleId;
+            Status = status;
         }
 
         public string FilePath { get; }
         public string Hash { get; }
         public int? Line { get; }
         public string Message { get; }
         public string ModuleKey { get; }
-        public SonarQubeIssueResolutionState ResolutionState { get; }
         public string RuleId { get; }
+        public string Status { get; }
```

Comment:
This is a bit cosmetic, but having a string Status does not feel nice, because it couples the rest of the code with a server-side value... Since we care only about Resolved/Unresolved states, perhaps a bool property would have been better. This way the magic strings would remain in the request object.

Reply:
I wanted the code to remain generic enough but I think the point is valid and will simplify current code .